### ISSUE TITLE
Angular Extend cannot extend DOMRect object with another object in Firefox...

### ISSUE THREAD

**Issue by mehmetotkun** [angular/angular.js#11232]
I have tried to extend an element `.getBoundingClientRect` result with another object. But, it cannot extended in FF.

Here is an pluker example [http://plnkr.co/edit/y5jNuiHCQVAHKNJElntO?p=preview](http://plnkr.co/edit/y5jNuiHCQVAHKNJElntO?p=preview)

**Comment by pkozlowski-opensource** [comment]
It is because FFox returns instances of [DOMRect](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsIDOMClientRect) objects, instead of regular JS objects. Unfortunately `DOMRect` don't allow us to enumerate keys thus we've got no way of knowing which properties of a given object should be copied. I'm afraid that you don't have other choice than coping properties of those special objects "by hand", ex.: http://plnkr.co/edit/7R2JB2vohrBWZBE2V8ef?p=preview

Closing this one for now as we don't want to special-case this in AngularJS. You might want to report it to FFox folks to make sure that they add a method of enumerating properites (ex.: `getKeys`) to those special objects.
